File: class-exception/python/src/iofilecls.py

import logging

class iofilecls:

	def read(self, filename):
		try:
			f = open(filename)
			content = f.read()

		except IOError:
			raise IOError('io error occured')
		finally:
			try:
				f.close()
			except:
				logging.error(f'{__name__} - cannot close file')
				# raise FileNotFoundError('filename not found')
				raise MyException('filename not found')


		print('end of func')
		return content


class MyException(Exception):

	def __init__(self, msg):
		self.msg = msg
		self.msg2 = msg

File: class-exception/c++/src/iofilecls.cpp

#include <string>
#include <fstream>
using namespace std;

#include "iofilecls.h"

string iofilecls::read(string filename) throw (MyException){

	ifstream fin (filename.c_str(), ios::in);
	string content = "";
	string line;
	if(fin){
		try{
			while(getline(fin, line)){
				content += line;
				content += string("\n");
			}
			fin.close();

		}
		catch(...){
			fin.close();
			throw MyException("read file error");
		}

	}
	else{
		// throw "FileNotFoundException";
		throw MyException("file not found");
	}


	return content;


}
